docs/proposals/20220603-kubernetes-vendor.md

```diff
@@ -0,0 +1,104 @@
+# Multi-runtime Support
+
+Authors:
+- [Ce Gao](https://github.com/gaocegege)
+
+## Summary
+
+envd builds and runs the environments with Docker. This proposal is to support Kubernetes as a runtime option. After this feature, envd can build the environment with Docker and run on Kubernetes.
+
+## Motivation
+
+Docker runtime only works for individual developers. There are many data scientists developing models on Kubernetes. In the current design, they may need to run `envd build`, then push the image to a OCI image registry manually. The image can be used after a deployment or a pod is created in Kubernetes.
+
+We can provide native support for ease of use.
+
+## Goals
+
+- Run environments on Kubernetes
+  - Compose Kubernetes resources (e.g. deployments, services, pods) when run `envd up`.
+  - Push the image to a registry.
+  - Port forward for sshd and jupyter notebook service.
+  - Sync the pod file system with the local host file system.
```

Comment:
What if the two file systems are not compatible?

Reply:
As we have set up port forwarding, how about using `rsync` instead.

Reply:
Actually I am for the idea of using `rsync`, given that it is mature and battle-tested.

Reply:
Oh I just realized that `syncthing` could be imported as a library, if that is more convenient I am okay with that 👍